src/store/accounts/api.ts

```diff
@@ -64,26 +65,26 @@ export const accountBalancesClient = atomFamilyWithQuery<
   }
 );
 
-export const accountBalancesBigNumber = atomFamily<
-  PrincipalWithNetworkUrl,
-  AccountBalanceResponseBigNumber
->(params =>
-  atom(get => {
-    const balances = get(accountBalancesClient(params));
-
-    const stxBigNumbers = Object.fromEntries(
-      accountBalanceStxKeys.map(key => [key, new BigNumber(balances.stx[key])])
-    ) as Record<AccountBalanceStxKeys, BigNumber>;
-
-    const stx: AccountStxBalanceBigNumber = {
-      ...balances.stx,
-      ...stxBigNumbers,
-    };
-    return {
-      ...balances,
-      stx,
-    };
-  })
+export const accountBalancesBigNumber = atomFamily<string, AccountBalanceResponseBigNumber>(
+  principal =>
+    atom(get => {
+      const network = get(currentNetworkState);
+
+      const balances = get(accountBalancesClient({ principal, networkUrl: network.url }));
```

Comment:
So, was the fix for network switching that this wasn't using the current network? It seems like there is a lot of refactoring in this commit, how would I have known how to debug/refactor this? Can you do a demo of how you debugged it and/or what led you to this solution?

Reply:
I'd love to show how I debugged this! I can try to describe in text, but something more interactive would be better imo. much of it was around enabling/disabling different components to determine where the issue was coming from. The ultimate issue was two things: [this diff here](https://github.com/blockstack/stacks-wallet-web/pull/1572/files#diff-8b6e56e8c1098538053b9d0c3063c7b9d5e373ae860b9c572179e329e4b278b5R88-R106) where there was an `async` before the `get`, creates a bug where because we return `localData` if present, the function does not (and should not) be async. This causes it to never resolve.

The other fix was adding `deepEqual` [here](https://github.com/blockstack/stacks-wallet-web/pull/1572/files#diff-57af48a8961860b9afe5173511351f10c2ba3f60766ff666be8ab416ba731d3eR89), [here](https://github.com/blockstack/stacks-wallet-web/pull/1572/files#diff-57af48a8961860b9afe5173511351f10c2ba3f60766ff666be8ab416ba731d3eR131), and [here](https://github.com/blockstack/stacks-wallet-web/pull/1572/files#diff-0e1aeb0aaf7de294ac796ec12ad49035931dea2f325239f63424276811d87701R126). Previously the built-in equality check was not sufficient in determining when atoms were using different values as params.

Reply:
Thx for the info ...would def be great to see a demo if you are available tomorrow at some point.

Reply:
I would be interested too !

Reply:
I'll try to find us some time (likely tomorrow)!